e2e/README-Subpath.md

```diff
@@ -0,0 +1,7 @@
+# Testing with subpath servers
+Some tests need multiple servers running in subpath mode. These tests have the cypress `Group: @subpath` metadata near the top of the test file. Instructions on running a server under subpath can be found here: [https://developers.mattermost.com/blog/subpath/](https://developers.mattermost.com/blog/subpath/)
+
+In the `cypress.json` configuration file, the `baseURL` setting will need to be updated with the subpath URL of the first server, and the `secondServerURL` setting with the subpath URL of the second server.
+
+### Running subpath tests on local machine
+Two mattermost servers running on the same machine must be served from different ports. To have the servers respond on the same URL and the same port under different subpaths, you will need to use a reverse proxy (nginx or apache) to proxy the same local url to both mattermost servers under different subpaths.
```

Comment:
I wonder if you could add more info on how to set this up, like a step-by-step to easily follow along by contributors (new and old) . The baseUrl is the typical local dev setup but the second server could be from the compiled version. Perhaps, a docker-compose that could easily bring up/down? Maybe just the base app and db of https://github.com/mattermost/mattermost-docker?

Reply:
Could you be more specific about what it is about setting up the second server that is unclear? I'd happy to clarify in the docs. The only difference between the two servers would be the siteURL setting and the DB config (we need to have two separate DBs because in one of the tests we check for the existence of user on one server but not on the other).
There are many ways to achieve this as well (actual config change vs environment variable), so I thought the link to the subpath setup would be sufficient.

I can copy my nginx reverse proxy config here if you think that would be helpful, but I intentionally left the specifics out because there are many ways to do it, and lots of documentation online on how to setup a reverse proxy. 

>Perhaps, a docker-compose that could easily bring up/down? Maybe just the base app and db of https://github.com/mattermost/mattermost-docker?

Are you suggesting a single docker compose file to bring up multiple mattermost servers and DBs, and a reverse proxy for subpath tests? This could take some time for me to figure out. I think it would be best tackled in a separate ticket.

Reply:
Spinning up a single subpath server or multiple servers in different port is easy but multiple servers on different subpaths in same URL and port is non-trivial. I agree on several ways to achieve it but it would be nice if a common setup guide is readily available for the purpose of E2E testing, especially in local dev. I mentioned docker-compose since I find it practical to do so but very much open to anything.

Reply:
@saturninoabril I've added docs with steps on how to set up the reverse proxy using nginx

Reply:
Awesome, thanks @ashishbhate!